torch/nn/modules/module.py

```diff
@@ -1165,7 +1166,7 @@ def _call_impl(self, *input, **kwargs):
             grad_fn = var.grad_fn
             if grad_fn is not None:
                 for hook in non_full_backward_hooks:
-                    wrapper = functools.partial(hook, self)
+                    wrapper = functools.partial(hook, weakref.proxy(self))
```

Comment:
You will have to use `weakref.ref()` and update the code a little to check that the ref is still valid to fix the CI.
You want to not use partial anymore and just define a local function I think.

Reply:
Not sure the backward hook have the same problem as they are created on the fly during the forward pass. So there is no ref cycle being created in this case.
Unless the user stores the output of the forward function on `self.` which should be rare enough that we're happy to make a cycle here.

So only the one for the load state dict would be enough.